competition/views.py

```diff
@@ -116,11 +126,27 @@ class ResultsView(DetailView):
 
     def get_context_data(self, **kwargs):
         context = super(ResultsView, self).get_context_data(**kwargs)
-
         context['categories'], context['teams'] = generate_results(self.object)
-
         return context
+    
+    def post(self,request,pk):
+        if request.user.is_staff:
+            self.object = self.get_object()
+            if self.request.POST['freeze']:
+                _, results = generate_results(self.object)
+                self.object.frozen_results = json.dumps(self.object.frozen_results)
+            else:
+                self.object.frozen_results = None
+            self.object.save()
+            return self.get(request,pk=pk)
 
+class PublicResultsView(ResultsView):
+    
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if self.object.frozen_results is None:
```

Comment:
Mozno sa mylim ale nemalo by tu byt praveze `is not None` ?

Reply:
Opravil som